plugins/vpp/ifplugin/nat_config.go

```diff
@@ -587,19 +550,59 @@ func (plugin *NatConfigurator) addDelAddressPool(addressPools []*nat.Nat44Global
 		} else if lastIP == nil {
 			lastIP = firstIP
 		}
+		if err = vppcalls.AddNat44AddressPool(firstIP, lastIP, addressPool.VrfId, addressPool.TwiceNat, plugin.vppChan, plugin.stopwatch); err != nil {
+			plugin.log.Error(err)
+			wasErr = err
+		}
+	}
 
-		// configure or remove address pool
-		if isAdd {
-			if err = vppcalls.AddNat44AddressPool(firstIP, lastIP, addressPool.VrfId, addressPool.TwiceNat, plugin.vppChan, plugin.stopwatch); err != nil {
-				plugin.log.Error(err)
-				wasErr = err
+	return wasErr
+}
+
+// Removes NAT address pool. Invalid address pool configuration is skipped with warning, configurator assumes that
+// such a data could not be configured to the vpp.
+func (plugin *NatConfigurator) delAddressPool(addressPools []*nat.Nat44Global_AddressPool) (err error) {
+	var wasErr error
+	for _, addressPool := range addressPools {
+		if addressPool.FirstSrcAddress == "" && addressPool.LastSrcAddress == "" {
+			// No address pool to remove
+			continue
+		}
+		var firstIP []byte
+		var lastIP []byte
+		if addressPool.FirstSrcAddress != "" {
+			firstIP = net.ParseIP(addressPool.FirstSrcAddress).To4()
+			if firstIP == nil {
+				// Do not return error here
+				plugin.log.Warnf("First address pool IP %s cannot be parsed and removed, skipping",
+					addressPool.FirstSrcAddress)
+				continue
 			}
-		} else {
-			if err = vppcalls.DelNat44AddressPool(firstIP, lastIP, addressPool.VrfId, addressPool.TwiceNat, plugin.vppChan, plugin.stopwatch); err != nil {
-				plugin.log.Error(err)
-				wasErr = err
+		}
+		if addressPool.LastSrcAddress != "" {
+			lastIP = net.ParseIP(addressPool.LastSrcAddress).To4()
+			if lastIP == nil {
+				// Do not return error here
+				plugin.log.Warnf("Last address pool IP %s cannot be parsed and removed, skipping",
+					addressPool.FirstSrcAddress)
```

Comment:
Should be `LastSrcAddress`.

Reply:
Also, should it really skip this loop with `continue` like it does with first ip?

Reply:
yes, it should. Address pool configuration requires both ends to be configured if they are set, otherwise, it returns an error. During removal, the behavior should be the same, otherwise it tries to remove non-existing address pool and vpp does not like it.